src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -410,7 +410,7 @@ private ProtocolToken ProcessBlob(int frameSize)
             _handshakeBuffer.Discard(frameSize);
 
             // Often more TLS messages fit into same packet. Get as many complete frames as we can.
-            while (_handshakeBuffer.ActiveLength > TlsFrameHelper.HeaderSize)
+            while (_handshakeBuffer.ActiveLength > TlsFrameHelper.HeaderSize && _framing != Framing.BeforeSSL3 && _framing == Framing.Unified)
```

Comment:
Why is the _framing != Framing.BeforeSSL3 check needed when we then require _framing == Framing.Unified?

Reply:
This is copy error and broke the optimization to process more frames together. I replaces it simpler `_framing == Framing.SinceSSL3`